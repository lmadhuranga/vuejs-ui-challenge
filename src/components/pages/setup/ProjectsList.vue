<style scoped>
  .panelSmall {
    position: fixed;
    top: 0;
    background: #fff;
    max-width: 400px;
    height: 100vh;
    right: 0;
  }

  .panelSmall.active:after {
    display: block;
    content: " ";
    position: fixed;
    width: 100%;
    height: 100%;
    background: rgba(15, 4, 35, 0.63);
    right: 400px;
    top: 0;
    z-index: -15;
  }

  .panelSmall .panelClose {
    position: absolute;
    margin-left: -46px;
    margin-top: 15px;
    padding: 7px 10px;
    background: #fff;
    border-radius: 100%;
  }

  [name="slectShift"] {
    width: 0;
    height: 0;
    position: absolute;
    visibility: hidden;
  }

  .helpText {
    margin-top: 20px;
    display: inline-block;
  }
</style>

<template>
  <div id="projectList">
    <!-- s panel small -->
    <div class="wall panelSmall active" style="display: block">
      <go-to-home></go-to-home>
      <h2>Project List</h2>

      <div class="accordionWrap">
        <project-list-menu :projects="projects" :selectedproject="selectedproject"></project-list-menu>
      </div>

      <!-- e accordion wrap -->
      <div class="panelFooter">
        <img src="panelFooter.jpg"/>
      </div>
    </div>
    <!-- e panel small -->
  </div>
  <!-- e #projectList-->
</template>

<script>
  import ProjectListMenu from "../../template/ProjectListMenu";
  import GoToHome from "../../template/GotoHome";

  export default {
    components: {
      GoToHome,
      ProjectListMenu
    },
    name: 'ProjectList',
    props: ['users', 'projects', 'selectedproject']

  }
</script>
