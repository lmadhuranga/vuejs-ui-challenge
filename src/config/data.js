export default {
  users: [
    {
      id: 1,
      name: 'Madhuranga Sendheera',
      address: 'Colombo, Sri Lanka',
      postCode: '12344',
      btwNumber: '12345',
      kvkNumber: '12345'
    },
    {
      id: 2,
      name: 'Randika Senadheera',
      address: 'Colombo, Sri Lanka',
      postCode: '12344',
      btwNumber: '12345',
      kvkNumber: '12345'
    },
  ],
  projects: {
    1: {
      id: 1,
      name: 'Project 1',
      shifts: [
        {
          id: 1,
          name: 'shift1',
          iban: 'IbanDetails',
          description: 'description',
          userId: 1
        },
        {
          id: 2,
          name: 'shift2',
          iban: 'IbanDetails',
          description: 'description',
          userId: 2
        }
      ]
    },
    2: {
      id: 1,
      name: 'Project 2'
    },
    3: {
      id: 1,
      name: 'Project 3'
    }
  }
}
