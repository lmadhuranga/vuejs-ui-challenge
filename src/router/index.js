import Vue from 'vue'
import Router from 'vue-router'
import Home from '@/components/pages/HomePage'
import UserAdd from '@/components/pages/UserAdd'
import UserView from '@/components/pages/UserView'
import UsersList from '@/components/pages/UsersList'
import SetupContainer from '@/components/pages/SetupContainer'
import Vuetify from 'vuetify'

Vue.use(Vuetify);
Vue.use(Router);

if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(plugin)
}

export default new Router({
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home
    },
    {
      path: '/userslist',
      name: 'usersList',
      component: UsersList
    },
    {
      path: '/setup',
      name: 'setup',
      component: SetupContainer,
      children: [
        {
          // UserProfile will be rendered inside User's <router-view>
          // when /user/:id/profile is matched
          path: 'add',
          component: UserAdd
        }]
    },
    {
      path: '/add',
      name: 'add',
      component: UserAdd
    },
    {
      path: '/:id',
      name: 'userView',
      component: UserView
    }
  ]
})

