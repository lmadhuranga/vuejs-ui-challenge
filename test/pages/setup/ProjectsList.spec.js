import Vue from 'Vue';
import ProjectsList from '../../../src/components/pages/setup/ProjectsList.vue'
import router from '../../../src/router'
import data from '../../../src/config/data'
import {shallow, mount} from "vue-test-utils"
import VueRouter from 'vue-router';

describe('ProjectsList.vue', () => {
  Vue.use(VueRouter);
  let cmp, users, projects, project, projectId;

  const createCmp = propsData => shallow(ProjectsList, {
    propsData,
    el: document.createElement('div'),
    router: router,
    render: h => h('router-view')
  });

  beforeEach(() => {
    users = data.users;
    projects = data.projects;
    project = projects[0];
    projectId = projects[0];
    cmp = createCmp({
      users: users,
      projects: projects,
      selectedproject:projectId
    });
  });

  it('should be able load projects', () => {
    expect(cmp.vm.projects).toEqual(projects);
  });

  it('should be able load projects', () => {
    expect(cmp.vm.users).toEqual(users);
  });

  it('should be able load projects', () => {
    expect(cmp.vm.selectedproject).toEqual(projectId);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  });
});
