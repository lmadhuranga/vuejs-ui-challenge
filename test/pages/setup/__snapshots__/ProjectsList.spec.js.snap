// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsList.vue has the expected html structure 1`] = `<!--  -->`;
